src/core/ext/filters/client_channel/http_proxy.cc

```diff
@@ -81,12 +73,31 @@ static char* get_http_proxy_server(char** user_cred) {
     proxy_name = nullptr;
   }
   gpr_free(authority_strs);
-done:
-  gpr_free(uri_str);
   grpc_uri_destroy(uri);
   return proxy_name;
 }
 
+grpc_uri* get_proxy_uri(char* proxy_env_var) {
```

Comment:
Can we pass the input as `const char*`?

Suggest naming the parameter as `proxy_env_var_name` or `proxy_env_var_key`.